### PR TITLE
Adds a philosophy section to README.md.

### PR DESCRIPTION
**Pull request by lexicalunit**
I was at [Abstractions.io Conference](http://abstractions.io/) in Pittsburgh this past weekend and I saw a talk by Nadia Odunayo about Code Hospitality. Here's a version of the talk presented at railsconf: https://www.youtube.com/watch?v=hHzWG1FltaE

I thought one of the suggestions she had could be an easy win for nanodbc: Add a Philosophy section to your project readme.

### CODE REVIEW

**Comment by mloskot** [comment]
Good idea

**Comment by mloskot** [comment]
I watched the lecture, good one.

File: README.md [hunk 1]
```diff
@@ -2,6 +2,20 @@
...
+
+The [latest C++ standards](https://isocpp.org/std/status) and [best practices](https://github.com/isocpp/CppCoreGuidelines) are _enthusiastically_ incorporated to make the library as future-proof as possible. To accommodate users who can not use the latest and greatest, [semantic versioning](http://semver.org/) and release notes will clarify required C++ features and/or standards for particular versions.
+
+### Design Decisions
+
+All complex objects in nanodbc follow the [pimpl (Pointer to IMPLementation)](http://c2.com/cgi/wiki?PimplIdiom) idiom to provide separation between interface and implementation, value semantics, and a clean `nanodbc.h` header file that includes nothing but standard C++ headers.
+
+Nanodbc wraps ODBC code, providing a simpler way to do the same thing. We try to be as featureful as possible, but I can't guarantee you'll never have to write supporting ODBC code. Personally, I have never had to do so.
```
Comment (mloskot) [inline README.md]: I don't want to be too anal about this, but isn't the lower-case-ness part of the "nanodbc" brand name?

Comment (lexicalunit) [inline README.md]: Good eye! The way I've always written it is that the project is named `nanodbc` but if it comes at the start of a sentence I've been capitalizing it. It does make more sense to be consistent though.

Comment (mloskot) [inline README.md]: OK

**Pull request merged**